virtual_ecosystem/models/abiotic_simple/microclimate.py

```diff
@@ -101,62 +100,47 @@ def run_microclimate(
             data=data,
             reference_data=data[var + "_ref"].isel(time_index=time_index),
             leaf_area_index_sum=leaf_area_index_sum,
-            layer_roles=layer_roles,
+            layer_structure=layer_structure,
             layer_heights=data["layer_heights"],
             upper_bound=upper,
             lower_bound=lower,
             gradient=gradient,
         ).rename(var)
 
     # Mean atmospheric pressure profile, [kPa]
-    output["atmospheric_pressure"] = (
-        (data["atmospheric_pressure_ref"])
-        .isel(time_index=time_index)
-        .where(output["air_temperature"].coords["layer_roles"] != "soil")
-        .rename("atmospheric_pressure")
-        .T
+    # VIVI: the previous code implied this should only be filled for atmospheric roles,
+    # but that didn't happen.
+    output["atmospheric_pressure"] = layer_structure.from_template()
+    output["atmospheric_pressure"][:] = data["atmospheric_pressure_ref"].isel(
+        time_index=time_index
     )
 
     # Mean atmospheric C02 profile, [ppm]
-    output["atmospheric_co2"] = (
-        data["atmospheric_co2_ref"]
-        .isel(time_index=0)
-        .where(output["air_temperature"].coords["layer_roles"] != "soil")
-        .rename("atmospheric_co2")
-        .T
-    )
+    # VIVI: This had a non varying time index
+    output["atmospheric_co2"] = layer_structure.from_template()
+    output["atmospheric_co2"] = data["atmospheric_co2_ref"].isel(time_index=time_index)
 
```

Comment:
yeah, I think I assumed constant CO2 when I set this up but you are right, that should be a time series, too.